otj-metrics-core/src/main/java/com/opentable/metrics/jvm/NmtMetrics.java

```diff
@@ -14,128 +14,72 @@
- * Furthermore, this is also the opportunity to ensure that we are reading the latest NMT values.  However, fetching
- * NMT information is non-trivial, so we want to rate-limit it.  Therefore, the design employed here is that on gauge
- * measurement, we check to update our NMT observations according to a refresh period.  Having refreshed the NMT
- * information, we can <em>also</em> check for further metrics to register.
- *
- * Cf. https://docs.oracle.com/javase/8/docs/technotes/guides/troubleshoot/tooldescr022.html
- *
- * In terms of thread-safety, we assume the worst, and handle {@link #register()} and the gauge functions being called
- * by arbitrary threads.
- *
- * FindBugs is bad at understanding the synchronized access and modification of {@link #nmt}, particularly when
- * referred to in lambdas, so we suppress {@code IS2_INCONSISTENT_SYNC} warnings.
  */
-@SuppressFBWarnings(value = "IS2_INCONSISTENT_SYNC",
-        justification = "FindBugs gets confused by the lambda references to nmt.")
 public class NmtMetrics {
-    private static final Logger LOG = LoggerFactory.getLogger(NmtMetrics.class);
     private static final Duration REFRESH_PERIOD = Duration.ofSeconds(10);
 
     private final String prefix;
     private final MetricRegistry metrics;
-    private final Set<String> registeredMetrics = new HashSet<>();
 
-    private Nmt nmt;
-    private Instant lastRefresh;
+    private final ScheduledExecutorService exec = Executors.newScheduledThreadPool(
+        1,
+        new ThreadFactoryBuilder()
+            .setNameFormat("nmt-fetcher-%d")
+            .build()
+    );
+
+    private final ConcurrentMap<String, AtomicLong> gauges = new ConcurrentHashMap<>();
 
     public NmtMetrics(final String metricNamePrefix, final MetricRegistry metrics) {
         prefix = metricNamePrefix;
         this.metrics = metrics;
     }
 
     public synchronized void register() {
-        nmt = Memory.getNmt();
-        if (nmt == null) {
-            LOG.info("got null NMT info; not registering any metrics");
-            return;
-        }
-        lastRefresh = Instant.now();
-        refreshCategories();
-        if (registeredMetrics.isEmpty()) {
-            LOG.warn("Didn't find any NMT metrics to register!");
-        }
-        registerGauges("total", () -> nmt.total);
+        exec.scheduleAtFixedRate(this::poll, 0, REFRESH_PERIOD.toMillis(), TimeUnit.MILLISECONDS);
     }
 
-    private void refreshCategories() {
-        final Set<String> atStart = ImmutableSet.copyOf(registeredMetrics);
-        nmt.categories.keySet().forEach(categoryName -> {
-            final String metricName = categoryName.toLowerCase(Locale.ROOT).replaceAll(" ", "-");
-            if (registeredMetrics.contains(metricName)) {
+    private void setGaugeValue(String metricName, long value) {
+        AtomicLong gauge = this.gauges.get(metricName);
+        if (gauge == null) {
```

Comment:
I guess I'm not completely convinced double reference check is that important here, but no matter